### ISSUE TITLE
Score definitions based on relevance for a given context (not complexity)

### ISSUE THREAD

**Issue by jonjiang1** [higgood/med-jargon-explain-inator#8]
AC:
- [ ] Given a list of definitions and a word and context, return a ranked list of "best" definition

**Comment by wammar** [comment]
@jonjiang1 any update on this?

**Comment by wammar** [comment]
@jonjiang1 any update on this?

**Comment by jonjiang1** [comment]
I'm currently stuck after looking into some options, it looks like Elastic and their vector db is a paid solution and I'm not entirely familiar enough with the space to know of another solution that would let us do ranked info retrieval easily.

**Comment by wammar** [comment]
ES offers a self-hosted solution for free: https://www.elastic.co/elastic-cloud-kubernetes. Alternatively, here's a relatively recent list of vector DBs: https://byby.dev/vector-databases but I'm not sure if any of them have a free tier.

Alternatively, you can implement a simplistic information retrieval system using tf/idf, so that the rest of the team is not blocked on this component, then we can decide whether it makes sense to upgrade it in a follow-up task.

@jonjiang1

**Comment by wammar** [comment]
We're pivoting to use LangChain as the solution for this. Jonathan and Bridger have a pair programming session scheduled on Saturday for this.

**Comment by wammar** [comment]
Some progress was made here, but it's not done.
